java/code/src/com/suse/manager/webui/websocket/WebsocketSessionConfigurator.java

```diff
@@ -35,8 +35,8 @@ public void modifyHandshake(ServerEndpointConfig config,
             HandshakeRequest request,
             HandshakeResponse response) {
         HttpSession httpSession = (HttpSession)request.getHttpSession();
-        HttpServletRequest sr = (HttpServletRequest) httpSession.getAttribute("__original_request__");
-        User user = new RequestContext(sr).getCurrentUser();
+        User user = Optional.ofNullable(httpSession.getAttribute("webUserID"))
+                .map(webUserIDObject ->UserFactory.lookupById((Long)webUserIDObject)).orElse(null);
```

Comment:
you can use `ifPresent` on `Optional` to replace the if below.

Reply:
you are completely right. I just changed the code. Thanks